apps/app/src/utils/pageScroll.ts

```diff
@@ -10,8 +17,10 @@ export const lockScroll = ({ mobileOnly }: { mobileOnly: boolean }) => {
 	}
 };
 
-export const unlockScroll = ({ mobileOnly }: { mobileOnly: boolean }) => {
-	document.body.style.paddingRight = "";
+export const unlockScroll = ({ mobileOnly, preventLayoutShift }: PropsType) => {
+	if (preventLayoutShift) {
```

Comment:
Po co jest ta opcja i kiedy chcemy ustawić ją na false? Wydaje mi się, że nigdy.

Reply:
W przypadku kiedy otwieramy menu mobilne dodawanie padding-right do body nic nie zmienia, ponieważ i tak nie widać tego przesunięcia strony spowodowanego usunięciem scrollbarów. Podobnie sprawa ma się w przypadku QuestionsSidebara w widoku mobilnym - cała strona jest zakryta po jego otwarciu

Natomiast dodanie tego odstępu może powodować w niektórych sytuacjach problemy np:
1. otwieram menu mobilne:

![image](https://user-images.githubusercontent.com/27455716/211064244-74c8d750-43d8-4dd4-a6af-82980ae72aed.png)

2. przekręcam widok na widok horyzontalny i oczom ukazuje się niepotrzebny odstęp od scrollbara:

![image](https://user-images.githubusercontent.com/27455716/211064437-f5edfd5c-9d0f-4bca-947b-55ee3e68e246.png)

Ta dodatkowa opcja została dodana właśnie z myślą o tym przypadku.

Faktem jest również to, że dodawanie odstępu od prawej po ukryciu scrollbara ma znaczenie tylko w przeglądarkach, gdzie on jest cały czas widoczny, czyli najczęściej na desktopowych urządzeniach / przeglądarkach. Na smartfonie tego niepożądanego efektu nie dostrzeżemy.

Reply:
To nie prościej byłoby po prostu usunąć ten padding przy obracaniu ekranu? :)

Reply:
Opisany wyżej przypadek można by było wyeliminować również w ten sposób, że w zależności od wartości `isAboweBreakpoint`, czyli krótko mówiąc w momencie przechodzenia z widoku mobilnego na desktopowy i odwrotnie jeśli menu było by otwarte to należało by dodać ten odstęp lub go zabrać, lecz problem jest taki że, jak scrollbary są widoczne to można sobie obliczyć ile tego odstępu dodać, ale jak chcę przywrócić ten odstęp i mieć nadal zablokowaną możliwość scrollowania to nie ma skąd tego odstępu obliczyć. Gdyby tak na przykład dać w useEffect warunek, że jeśli np sidebar jest otwarty i `isAboreBreakpoint === true` to dodaj scrollbary i usuń odstęp to by załatwiło sprawę, ale wpływ na to czy maja być widoczne scrollbay mają rówmnież inne elementy np menu i modale. To często generowałoby konflikty i doprowadzało by do sytuacji, że np menu mobilne ma widoczne scrollbary, a nie powinno.

Krótko mówiąc ta dodatkowa właściwość eliminuje konflikty w bardzo specyficznych sytuacjach.

Reply:
mając nadal otwarte menu w widoku mobilnym. To trochę skomplikowane

Reply:
https://github.com/typeofweb/devfaq/pull/484#discussion_r1061991079 spełnienie wymagania z tego komentarza, generuje najwięcej problemów

Reply:
@mmiszy tak z ciekawości zerknąłem, jak to jest rozwiązane na allegro.pl. W wersji mobilnej, jak otworzę sidebar 'FIltry i kategorie' to do `body` jest dodawana klasa css z `overflow: hidden; `, ale jak z otwatym sidebarem zmienie szerokość viewportu do rozmiaru desktopowego to nadal mam zablokowaną możliwość scrollowania 😁

Zawsze jakieś kompromisy...

Reply:
ale przedstawię zaraz jeszcze jedna wersję chyba ostateczną, jeśli nie ona to już sam niewiem...

Reply:
Dobra, próbowałem przygotować inną wersję, w której stan czy menu jest otwarte, czy zamknięte oraz funkcje zmieniające ten stan znajdowałyby się w `UIProvider`, ale kod zaczyna wyglądać strasznie, a każdy element: menu, modale, sidebar ma funkcjonalność opisywaną praktycznie w sposób okropny, bo trzeba np. w modalu brać pod uwagę czy menu jest otwarte / zamknięte tak samo z sidebarem....

Ja mam taką propozycję:
- zostaje taka wersja jaka jest - wiem są kompromisy, ale nie da się ich w 'sensowny' sposób przeskoczyć (przez sensowny mam na myśli nie opisywać, jak dany element ma się zachować każdego szczególnego przypadku w zależności od stanu innego elementu na stronie)
- idziemy na kompromis i każde przejście z mobile na desktop bądź na odwrót powoduje 'wyzerowanie' stanu tego element do ustawień początkowych np. otwarte menu mobilne przy przejściu do widoku desktopowego jest zamykane i po ponownym powrocie do widoku mobilnego nadal będzie zamknięte.

Na chwilę obecną przetestowałem już tyle wersji, podpatrywałem, jak sobie z tym radzą inne serwisy i nie da się tego zrobić inaczej. Zawsze jest jakiś kompromis, chyba że ręcznie rozpiszesz i odsłużysz wszystkie możliwe kombinacje otwarcia i zamknięcia menu, modali, sidebarów. Współczuję jednak jak postanowisz dodać kolejny element...